### PR TITLE
fix(logging): improve java core's debug logging

### PR DESCRIPTION
**Pull request by padamstx**
This commit adds some debug logging in various
components, and also modifies some tests so that
it's easy to enable debug logging when running
individual sets of tests.

### CODE REVIEW

File: src/test/java/com/ibm/cloud/sdk/core/test/service/BaseServiceTest.java [hunk 1]
```diff
@@ -46,7 +49,16 @@
  * Unit tests associated with the BaseService core class.
  *
  */
-public class BaseServiceTest {
+public class BaseServiceTest extends UnitTestBase {
+
+  // Logging level used by this test.
+  // For debugging, set this to Level.FINE or Level.ALL, etc.
+  private static Level logLevel = Level.FINE;
```
Comment (pyrooka) [inline src/test/java/com/ibm/cloud/sdk/core/test/service/BaseServiceTest.java:56]:
```suggestion
  private static Level logLevel = Level.SEVERE;
```
? :)

**Review by pyrooka — APPROVED** [review]
LGTM!

File: src/test/java/com/ibm/cloud/sdk/core/test/service/BaseServiceTest.java (hunk 1 above)
Comment (padamstx) [reply src/test/java/com/ibm/cloud/sdk/core/test/service/BaseServiceTest.java:56]: oops

Comment (padamstx) [reply src/test/java/com/ibm/cloud/sdk/core/test/service/BaseServiceTest.java]: fixed

File: src/main/java/com/ibm/cloud/sdk/core/http/HttpClientSingleton.java [hunk 2]
```diff
@@ -352,7 +351,6 @@ private OkHttpClient reconfigureClientInterceptors(OkHttpClient client, String i
       for (Iterator<Interceptor> iter = builder.interceptors().iterator(); iter.hasNext(); ) {
         Interceptor element = iter.next();
         if (interceptorToRemove.equals(element.getClass().getSimpleName())) {
-          LOG.log(Level.FINE, "Removing interceptor " + element.getClass().getName() + " from http client instance.");
```
Comment (dpopp07) [inline src/main/java/com/ibm/cloud/sdk/core/http/HttpClientSingleton.java:355]: I'm curious - why remove these logs?

**Review by dpopp07 — APPROVED** [review]
Looks good! 👍

File: src/main/java/com/ibm/cloud/sdk/core/http/HttpClientSingleton.java (hunk 2 above)
Comment (padamstx) [reply src/main/java/com/ibm/cloud/sdk/core/http/HttpClientSingleton.java:355]: I think I just didn't find those all that useful.  That's my story and I'm sticking to it.   I can easily add them back if needed.

Comment (dpopp07) [reply src/main/java/com/ibm/cloud/sdk/core/http/HttpClientSingleton.java:355]: Sounds good haha, just wanted to understand the justification. I think that's sufficient 🙂

**Comment by padamstx** [comment]
Temporarily changed this PR to draft status while I'm working on adding the redact secrets function and simplifying the testcase changes I original made in this PR.

**Comment by padamstx** [comment]
@dpopp07 @pyrooka,
I've updated the PR to include a new logging interceptor (HttpLogger) which mimics the behavior of the okhttp3 HttpLoggingInterceptor class, except that it redacts secrets within the debug messages that it logs.

I've also made it easier to enable debug logging...
The easiest way to do this while running tests is like this (in case you wanted to try it out):
```
cd <project-root>
mvn test -Dtest=<testclassname> -Djava.util.logging.config.file=debug-logging.properties

e.g.
mvn test -Dtest=ContainerAuthenticatorTest -Djava.util.logging.config.file=debug-logging.properties
```
(I added debug-logging.properties to the project in this PR).

**Review by dpopp07 — APPROVED** [review]
Updates look good to me 👍

**Review by pyrooka — APPROVED** [review]
Still LGTM!

**Pull request merged**